geoapps/inversion/electricals/direct_current/pseudo_three_dimensions/driver.py

```diff
@@ -0,0 +1,152 @@
+#  Copyright (c) 2022 Mira Geoscience Ltd.
+#
+#  This file is part of geoapps.
+#
+#  geoapps is distributed under the terms and conditions of the MIT License
+#  (see LICENSE file at the root of this source code package).
+
+import os
+import sys
+from copy import deepcopy
+
+import numpy as np
+from geoh5py.data import Data
+from geoh5py.groups.simpeg_group import SimPEGGroup
+from geoh5py.workspace import Workspace
+
+from geoapps.inversion.components.data import InversionData
+from geoapps.inversion.components.windows import InversionWindow
+from geoapps.inversion.electricals.direct_current.pseudo_three_dimensions.constants import (
+    validations,
+)
+from geoapps.inversion.electricals.direct_current.pseudo_three_dimensions.params import (
+    DirectCurrentPseudo3DParams,
+)
+from geoapps.inversion.electricals.direct_current.two_dimensions.params import (
+    DirectCurrent2DParams,
+)
+from geoapps.inversion.line_sweep.driver import LineSweepDriver
+from geoapps.shared_utils.utils import get_locations, weighted_average
+from geoapps.utils.models import get_drape_model
+from geoapps.utils.surveys import extract_dcip_survey
+
+
+class DirectCurrentPseudo3DDriver(LineSweepDriver):
+
+    _params_class = DirectCurrentPseudo3DParams
+    _validations = validations
+
+    def __init__(self, params: DirectCurrentPseudo3DParams):  # pylint: disable=W0235
+        super().__init__(params)
+        lookup = self.get_lookup()
+        self.write_files(lookup)
+        if params.files_only:
+            sys.exit("Files written")
+
+    def write_files(self, lookup):
+        """Write ui.geoh5 and ui.json files for sweep trials."""
+
+        forward_only = self.pseudo3d_params.forward_only
+        ifile = DirectCurrent2DParams(forward_only=forward_only).input_file
+
+        with self.workspace.open(mode="r+"):
+
+            window = InversionWindow(self.workspace, self.pseudo3d_params)
+            data = InversionData(self.workspace, self.pseudo3d_params, window.window)
+
+            xyz_in = get_locations(self.workspace, self.pseudo3d_params.mesh)
+            models = {"starting_model": self.pseudo3d_params.starting_model}
+            if not forward_only:
+                models.update(
+                    {
+                        "reference_model": self.pseudo3d_params.reference_model,
+                        "lower_bound": self.pseudo3d_params.lower_bound,
+                        "upper_bound": self.pseudo3d_params.upper_bound,
+                    }
+                )
+
+            for uuid, trial in lookup.items():
+
+                status = trial.pop("status")
+                if status == "pending":
```

Comment:
Just for clarity, can we just check for the non-pending case and continue instead. Will avoid the super long indentation.

Reply:
done